### DISCUSSION TITLE
Translations (Weblate) - Strings reappear as untranslated

### DISCUSSION THREAD

**Discussion by ERYpTION** [osmandapp/OsmAnd#21014] [Q&A]
Doing OsmAnd translations in Weblate I have recently noticed that some strings (about 170, iOS) reappear as untranslated despite having been correctly translated recently. It says "String updated in the repository" (by user None) for each of these strings in the History tab. Should I add the same translations again? It's a bit tedious having to do this once again but I assume I have to do it so the text doesn't appear untranslated in the app?

**Comment by dmpr0** [comment]
Hi! 
Could you please provide an example? We recently remove duplicates of English strings from localized files, now they can be translated.

**Comment by ERYpTION** [comment]
I think it is all the currently untranslated strings for Danish in the iOS section: For example: https://hosted.weblate.org/translate/osmand/ios/da/?offset=100&q=state%3A%3Ctranslated&sort_by=-priority%2Cposition&checksum=#history

**Reply by ERYpTION** [reply]
Did the translations again. Hopefully for the last time.

**Comment by MarioSwitch** [comment]
@dmpr0 Same for French, there are 184 strings uncompleted, but 128 of them are marked as "Failed check: already translated"...
![image](https://github.com/user-attachments/assets/965f1441-bfde-4305-823e-bfdb1a2ec9a9)

Some of them have been translated many times but translations are reverted automatically by Weblate... I think there may be a conflict between Weblate and iOS repo 🤔
![image](https://github.com/user-attachments/assets/17834e57-b039-4695-9b91-e2d63afbf1db)

**Comment by ZeljkoBG** [comment]
Weblate is bad. Migrate somewhere! Anywhere.

**Reply by MarioSwitch** [reply]
Why?

**Reply by ZeljkoBG** [reply]
If you have translated on other web translation services, then you understand.
"Why?" - well here is one reason. OsmAnd stuff have no influence and do not manage the translation of their application. I have an example of Serbian Cyrillic! The translating member has desecrated the translation with a non-existent and wrong translation...

**Reply by MarioSwitch** [reply]
> I have an example of Serbian Cyrillic! The translating member has desecrated the translation with a non-existent and wrong translation...

What's the link with Weblate? Any other open-source translation platform would have the same problem 🤔

**Reply by ZeljkoBG** [reply]
"open-source translation platform" - why is it necessary for OsmAnd to go on such a platform? They don't have the money for something more professional or won't give when they can free :)

If the app is free, then the platform is free. If they earn, they have to give.

**Reply by MarioSwitch** [reply]
> why is it necessary for OsmAnd to go on such a platform?

To have their app translated in many languages!?

**Reply by MarioSwitch** [reply]
> If the app is free, then the platform is free.

Weblate offers free hosting for open-source apps

**Reply by vshcherb** [reply]
I didn't get that impression weblate is bad. It's being 10 years may be we're using weblate ofcourse it's possible to switch to alternatives though still if we can fix it with them it's good cause it supports  (weblate opensource itself) opensource products

**Comment by ERYpTION** [comment]
Happened again. 170 strings in iOS now untranslated again.

**Reply by MarioSwitch** [reply]
I confirm: 184 untranslated in French, of which 181 marked as "Already translated"
